### PR TITLE
Align the version of jdt.core in maven to the one from gradle

### PR DESCRIPTION
**Pull request by holgerschill**
Signed-off-by: Holger Schill <Holger.Schill@itemis.de>

### CODE REVIEW

File: releng/org.eclipse.xtext.tycho.parent/pom.xml [hunk 1]
```diff
@@ -25,8 +25,8 @@
 		<log4j.version>1.2.16</log4j.version>
 		<commons-logging.version>1.1.3</commons-logging.version>
 		<commons-cli.version>1.2</commons-cli.version>
-		<equinox-common-version>3.6.0</equinox-common-version>
-		<jdt-core-version>3.10.0</jdt-core-version>
+		<equinox-common-version>3.8.0</equinox-common-version>
+		<jdt-core-version>3.12.2</jdt-core-version>
```
Comment (spoenemann) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: I don't think these versions are used anywhere. The Tycho build does not have any Maven dependencies. There's a separate parent pom for the Maven plugin build.

Comment (holgerschill) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: Sorry I do not understand what you mean.

Comment (spoenemann) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: I think these properties are leftovers from the old Maven build before the repo splitting. The xtext-eclipse build is purely based on Tycho and a target platform, so all dependencies are resolved via p2 repositories. That means these properties shouldn't be used anywhere in the Maven build (please verify via text search).

Comment (holgerschill) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: That would imply that when we decide to not change the MANIFEST.MF files we cannot do something in maven, right?

Comment (holgerschill) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: Indeed the properties are not used.

Comment (spoenemann) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: There are still Maven builds without Tycho in xtext-maven and xtext-xtend. There the properties are still relevant. But for Tycho builds the dependencies are determined via each bundle's MANIFEST.MF and the common target platform.

Comment (holgerschill) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: I know but this does not answer the question if we can do something if we do not want to change the MANIFEST.MF.

Comment (spoenemann) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: What do you mean by "do something"?

Comment (holgerschill) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: I mean curve the version in stone without changing the MANIFEST.MF files just for the build. Currently it depends on the target-platform.

Comment (spoenemann) [inline releng/org.eclipse.xtext.tycho.parent/pom.xml]: You mean we shouldn't allow the dependencies to diverge between Maven and the target platform? Maybe we could solve that by re-running the xtext-extras tests in the eclipse build using the target platform, which would verify whether our bundles are binary compatible with Eclipse Luna. That could run as an extra pipeline stage in the Jenkins build.

**Pull request closed without merging**